ci/run_ci.sh

```diff
@@ -0,0 +1,105 @@
+
+#!/bin/bash
+
+#set -x
+
+
+#smoke test environment specifications:
+#for scale perf
+export SCALE=6
+export RUNS=1
+export TIMEOUT=10
+#for kube burner 
+export JOB_ITERATIONS=3
+export NODE_COUNT=2
+export PODS_PER_NODE=2
+#for upgrade perf
+export VERSION=4.6.12
+#for router perf
+export PBENCH_SERVER='pbench.dev.openshift.com'
+export COMPARE=false
+export HTTP_TEST_SUFFIX='smoke-test'
+export HTTP_TEST_SMOKE_TEST=true
+
+
+#pull requests list
+git ls-remote origin 'pull/*/head' | awk '{print $2}' |
+while read ref; do
+  pr=$(echo $ref | cut -d/ -f3)
+  git fetch origin $ref > /dev/null
+  files_changed=$(git show --pretty=format:'' --name-only FETCH_HEAD|wc -l)
+  echo "PR number $pr has changes in $files_changed files"
+done
+
+
+#git checkout /pr/<pr-no>   the pr number can be accepted as input or obtained from pr_list
+
+diff_list=`git diff --name-only origin/master`
+
+
+if [ -z "$diff_list" ]
+then
+      echo "No changes noticed , exiting"
+      exit 0
+else
+      echo -e " List of files changed : ${diff_list} "
+fi
+
+cat > results.markdown << EOF
+Results for e2e-benchmarking CI Test
+Test                                             | Result | Runtime  |
+-------------------------------------------------|--------|----------|
+EOF
+
+test_list=`find ./workloads/ -maxdepth 2 -name '*t.sh' -type f `
+
+echo -e  "$test_list"       #test_list contains the list of scripts in all workloads
+
+
+
+for i in ${test_list}; do            
+
+start_time=`date`
+
+WORDTOREMOVE="./workloads/"
+FOO=${i//$WORDTOREMOVE/}                    #shell action to remove ./workloads/ prefix from the test_list entries
+
+directory=`expr match "$FOO" '\(.*/\)'`     #to extract the workload directory name 
+command=`expr match "$FOO" '.*\(run.*\)'`   #to extract the shell script name to run
+echo $directory $command
+
+if [[ ( $directory == network-perf/ ) ]]; # && ( $command =~ "git" ) ]];    #ignore all network-perf tests as they do not terminate
+then 
+    continue
+fi 
+
+figlet "CI test for ${FOO}"                 # figlet gives a heading for each test
+
+cd workloads/
+cd $directory
+echo $PWD
+
+(sleep 500s; sudo pkill -f $command) &      #to kill a workload script if it doesn't execute within 500s ; runs in the background
+
+bash $command                               #to execute each shell script
+
+EXIT_STATUS=$?
+
+if [ "$EXIT_STATUS" -eq "0" ]               #to check if the workloads exit successfully or fail
+then
+    result="PASS"
+else
+    result="FAIL"
+fi
+end_time=`date`
+duration=`date -ud@$(($(date -ud"$end_time" +%s)-$(date -ud"$start_time" +%s))) +%T`
+cd ..
```

Comment:
No need to change directories here. Directory print the results.markdown

Reply:
no i have to go back from the directory because i had entered the directory before to run the workload scripts